src/Downloader/Args.cs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+public struct Args
+{
+    public string Org { get; set; }
+    public List<string> Repos { get; set; }
+    public string GithubToken { get; set; }
+    public string? IssueDataPath { get; set; }
+    public int? IssueLimit { get; set; }
+    public string? PullDataPath { get; set; }
+    public int? PullLimit { get; set; }
+    public int? PageSize { get; set; }
+    public int? PageLimit { get; set; }
+    public int[] Retries { get; set; }
+    public Predicate<string> LabelPredicate { get; set; }
+    public bool Verbose { get; set; }
+
+    static void ShowUsage(string? message = null)
+    {
+        string executableName = Process.GetCurrentProcess().ProcessName;
+
+        Console.WriteLine($$"""
+            ERROR: Invalid or missing arguments.{{(message is null ? "" : " " + message)}}
+
+            Usage:
+              {{executableName}} --repo {org/repo1}[,{org/repo2},...] --label-prefix {label-prefix} [options]
+
+              Required arguments:
+                --repo              The GitHub repositories in format org/repo (comma separated for multiple).
+                --label-prefix      Prefix for label predictions. Must end with a character other than a letter or number.
+
+              Required for downloading issue data:
+                --issue-data        Path for issue data file to create (TSV file).
+
+              Required for downloading pull request data:
+                --pull-data         Path for pull request data file to create (TSV file).
+
+              Optional arguments:
+                --issue-limit       Maximum number of issues to download.
+                --pull-limit        Maximum number of pull requests to download.
+                --page-size         Number of items per page in GitHub API requests.
+                --page-limit        Maximum number of pages to retrieve.
+                --retries           Comma-separated retry delays in seconds. Default: 30,30,300,300,3000,3000.
+                --token             GitHub access token. Default: Read from GITHUB_TOKEN env var.
+                --verbose           Enable verbose output.
+            """);
+
+        Environment.Exit(1);
+    }
+
+    public static Args? Parse(string[] args)
+    {
+        Args config = new()
+        {
+            Retries = [30, 30, 300, 300, 3000, 3000]
+        };
+
+        Queue<string> arguments = new(args);
+        while (arguments.Count > 0)
+        {
+            string argument = arguments.Dequeue();
```

Comment:
I'm curious whether you considered using System.CommandLine functionality, and if yes - what were the reason to discard it?

Reply:
Definitely did. And I do expect to adopt S.CL at some point. We've been working on a pending new version of S.CL and I want to wait for it to be ready, and then that'll also give us a baseline of whether adopting it affects the live labeling speed/cost we observe after rollout.